pkg/controllers/user/alert/watcher/pod.go

```diff
@@ -156,86 +155,76 @@ func (w *PodWatcher) watchRule() error {
-				data["restart_times"] = strconv.Itoa(alert.Spec.PodRule.RestartTimes)
-				data["restart_interval"] = strconv.Itoa(alert.Spec.PodRule.RestartIntervalSeconds)
-
-				if details != "" {
-					data["logs"] = details
-				}
+			clusterDisplayName := common.GetClusterDisplayName(w.clusterName, w.clusterLister)
+			projectDisplayName := common.GetProjectDisplayName(alert.Spec.ProjectName, w.projectLister)
 
-				workloadName, err := w.getWorkloadInfo(pod)
-				if err != nil {
-					logrus.Warnf("Failed to get workload info for %s:%s %v", pod.Namespace, pod.Name, err)
-				}
-				if workloadName != "" {
-					data["workload_name"] = workloadName
-				}
+			data := map[string]string{}
+			data["rule_id"] = ruleID
+			data["group_id"] = alert.Spec.GroupName
+			data["alert_name"] = alert.Spec.DisplayName
+			data["alert_type"] = "podRestarts"
+			data["severity"] = alert.Spec.Severity
+			data["cluster_name"] = clusterDisplayName
+			data["project_name"] = projectDisplayName
+			data["namespace"] = pod.Namespace
+			data["pod_name"] = pod.Name
+			data["container_name"] = containerStatus.Name
+			data["restart_times"] = strconv.Itoa(alert.Spec.PodRule.RestartTimes)
+			data["restart_interval"] = strconv.Itoa(alert.Spec.PodRule.RestartIntervalSeconds)
 
-				if err := w.alertManager.SendAlert(data); err != nil {
-					logrus.Debugf("Error occurred while getting pod %s: %v", alert.Spec.PodRule.PodName, err)
-				}
+			if details != "" {
+				data["logs"] = details
 			}
 
-			return
+			workloadName, err := w.getWorkloadInfo(pod)
+			if err != nil {
+				logrus.Warnf("Failed to get workload info for %s:%s %v", pod.Namespace, pod.Name, err)
+			}
+			if workloadName != "" {
+				data["workload_name"] = workloadName
+			}
+
+			if err := w.alertManager.SendAlert(data); err != nil {
+				logrus.Debugf("Error occurred while getting pod %s: %v", alert.Spec.PodRule.PodName, err)
+			}
 		}
+
+		return
 	}
 
 }
 
 func (w *PodWatcher) getRestartTimeFromTrack(alert *v3.ProjectAlertRule, curCount int32) int32 {
 	name := alert.Name
 	namespace := alert.Namespace
-
-	obj, ok := w.podRestartTrack.Load(namespace + ":" + name)
-	if !ok {
-		return curCount
-	}
-	tracks := obj.([]restartTrack)
-
 	now := time.Now()
-
-	if len(tracks) == 0 {
-		tracks = append(tracks, restartTrack{Count: curCount, Time: now})
-		w.podRestartTrack.Store(namespace+":"+name, tracks)
-		return curCount
-	}
-
-	for i, track := range tracks {
-		if now.Sub(track.Time).Seconds() < float64(alert.Spec.PodRule.RestartIntervalSeconds) {
-			tracks = tracks[i:]
-			tracks = append(tracks, restartTrack{Count: curCount, Time: now})
-			w.podRestartTrack.Store(namespace+":"+name, tracks)
-			return track.Count
+	currentRestartTrack := restartTrack{Count: curCount, Time: now}
+	currentRestartTrackArr := []restartTrack{currentRestartTrack}
+
+	obj, loaded := w.podRestartTrack.LoadOrStore(namespace+":"+name, currentRestartTrackArr)
+	if loaded {
+		tracks := obj.([]restartTrack)
```

Comment:
Looks like formatting is off. It is more than 4 spaces

Reply:
Hi Alena, it's 4 spaces, may the web preview looks like more than 4 spaces